From: Philip Withnall <withnall@endlessm.com>
Date: Mon, 28 Oct 2019 19:40:26 +0000
Subject: gdbusauthmechanismsha1: Create .dbus-keyrings directory recursively

If the directory is overridden, for example when running tests, the
parent directory of `.dbus-keyrings` (i.e. the fake `$HOME` directory)
might not exist. Create it automatically.

This should realistically not have an effect on non-test code.

Signed-off-by: Philip Withnall <withnall@endlessm.com>
Bug: #1912
Origin: upstream, 2.63.1, commit:9df8d76c972de7cad6e62f7fbdda043ce86b8cc7
---
 gio/gdbusauthmechanismsha1.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/gdbusauthmechanismsha1.c b/gio/gdbusauthmechanismsha1.c
index d29048a..2754d3c 100644
--- a/gio/gdbusauthmechanismsha1.c
+++ b/gio/gdbusauthmechanismsha1.c
@@ -292,7 +292,7 @@ ensure_keyring_directory (GError **error)
       goto out;
     }
 
-  if (g_mkdir (path, 0700) != 0)
+  if (g_mkdir_with_parents (path, 0700) != 0)
     {
       int errsv = errno;
       g_set_error (error,
